pkg/lockfile/parse-yarn-lock.go

```diff
@@ -137,6 +146,43 @@ func determineYarnPackageVersion(group []string) string {
 	return ""
 }
 
+// You can find the line parsing regex in action here: https://regex101.com/r/QoJ3b7/3
+func determineYarnPackageDependencies(group []string) []YarnDependency {
+	indentCount := -1
+	results := make([]YarnDependency, 0)
+	lineParsing := cachedregexp.MustCompile(`^"?(?P<package_name>[^\s":]+)"?\s*:?\s*"?(?P<targeted_version>[^"\n]+)"?$`)
+
+	for _, line := range group {
+		cleaned := strings.TrimSpace(line)
+		if strings.HasPrefix(cleaned, "dependencies") {
+			// start of the dependencies section
+			indentCount = len(line) - len(cleaned)
+		} else if indentCount != -1 && len(line)-len(cleaned) == indentCount {
+			// end of the dependencies section, we can stop there
+			break
+		} else if indentCount != -1 {
+			// A line inside the dependencies section, lets parse it
+			match := lineParsing.FindStringSubmatch(cleaned)
+			name, version := match[1], match[2]
+			var registry string
+
+			if strings.Contains(version, ":") {
+				registry, version, _ = strings.Cut(version, ":")
+			} else {
+				registry = "npm"
+			}
```

Comment:
maybe comment the expected format? also, since the 3rd value of `Cut` returns whether or not `sep` was found, why not leverage that and avoid a call to `Contains` before?
```suggestion
			registry, version, found := strings.Cut(version, ":")
			if !found {
				registry = "npm"
			}
```